internal/dcache/rpc/server/handler.go

```diff
@@ -119,9 +115,13 @@ type mvInfo struct {
 	// Both StartSync and EndSync will quiesce IOs just before they move the mv into and out of syncing state, and
 	// resume IOs once the MV is safely moved into the new state.
 	//
-	// This boolean flag will be set by StartSync and EndSync to put the mv in quiesce state and it'll be read
-	// and honored by the various chunk related APIs - GetChunk, PutChunk, etc.
-	quiesceIOs atomic.Bool
+	// opMutex is used to ensure that only one operation, chunk IO (get, put or remove chunk) or
+	// sync (start sync or end sync) is in progress at a time.
+	// IO operations like get, put or remove chunk takes read lock on opMutex, and sync operations
+	// like StartSync or EndSync takes write lock on it.
+	// This ensures that the sync operation waits for the ongoing IO operations to complete.
+	// It also makes sure that no new IO operations till the sync is complete.
```

Comment:
It also makes sure that if start/end sync is waiting for the write lock, no new IO operations are started till the start/end sync gets the write lock and completes. This ensures that a continuous flow of IOs will not delay the start/end sync indefinitely.

Reply:
Updated